vllm/model_executor/models/phi3v.py

```diff
@@ -389,15 +390,61 @@ def _apply_hf_processor(
         processed_outputs['input_ids'] = token_ids
         return processed_outputs
 
-    def _get_dummy_mm_kwargs(
+    def _get_prompt_replacements(
+        self,
+        mm_items: MultiModalDataItems,
+        hf_inputs: BatchFeature,
+        mm_processor_kwargs: Mapping[str, object],
+    ) -> list[PromptReplacement]:
+        hf_processor = self._get_hf_processor()
+        image_tokens: list[str] = hf_processor.img_tokens  # type: ignore
+
+        mm_config = self.ctx.get_mm_config()
+        max_images = mm_config.limit_per_prompt.get("image", 1)
+
+        def get_replacement_phi3v(item_idx: int):
+            image_size = mm_items.get_image_size(item_idx)
+            num_tokens = get_phi3v_image_feature_size(
+                input_width=image_size.width,
+                input_height=image_size.height,
+                num_crops=mm_processor_kwargs.get("num_crops", 16),
+            )
+
+            return [_IMAGE_TOKEN_ID] * num_tokens
+
+        return [
+            PromptReplacement(
+                modality="image",
+                target=image_token,
+                replacement=get_replacement_phi3v,
+            ) for image_token in image_tokens[:max_images]
+        ]
```

Comment:
I think this won't trigger prompt replacement, because the placeholder like `<|image_1|>` etc (the image_tokens of processor) has been replaced to `<|image|>*num_img_tokens` (the special_image_token of processor) during processor calling.

Reply:
True. Nevertheless we will use `replacement` for finding the placeholder locations from the HF-processed tokens.

Reply:
Let's keep this as it summarizes what HF processor does.